src/sdkV2.ts

```diff
@@ -408,6 +425,77 @@ export default class CegaEvmSDKV2 {
    * USER FACING METHODS
    */
 
+  // ====== lpCega Offramp (Vault Token Market) code starts here
+
+  async getOfframpFeeBps(): Promise<number> {
+    const cegaEntry = await this.loadCegaEntry();
+    return cegaEntry.getOfframpFeeBps();
+  }
+
+  async getOrderData(maker: EvmAddress, orderHash: string): Promise<GetOrderDataResponse> {
+    const cegaEntry = await this.loadCegaEntry();
+    return cegaEntry.getOrderData(maker, orderHash);
+  }
+
+  async hashOrder(order: LpCegaOfframpOrder): Promise<string> {
+    const cegaEntry = await this.loadCegaEntry();
+    return cegaEntry.hashOrder(order);
+  }
+
+  async signOrder(cegaEntry: ethers.Contract, order: LpCegaOfframpOrder): Promise<string> {
+    const signer = this._signer;
+    const network = await this._provider.getNetwork();
+
+    if (!signer) {
+      throw new Error('Signer not defined');
+    }
+
+    const domain = {
+      name: 'Cega Offramp Entry',
+      version: '1',
+      chainId: network.chainId,
+      verifyingContract: cegaEntry.address,
+    };
+
+    return (signer as ethers.providers.JsonRpcSigner)._signTypedData(
+      domain,
+      TYPES_OFFRAMP_ORDER,
+      order,
+    );
+  }
+
+  async fillOrder({
+    order,
+    swapMakingAmount,
+  }: FillOrderParams): Promise<ethers.providers.TransactionResponse> {
+    const cegaEntry = await this.loadCegaEntry();
+    const makerSig = await this.signOrder(cegaEntry, order);
```

Comment:
the reason you can't use this is because the `makerSig` is the signature by the signer upon creating the order

vs the `fillOrder` is done by the taker that is fulfilling the order

so these two functions should not be coupled